site/docs/api/serialization.md

```diff
@@ -315,9 +316,58 @@ hazelcast:
         "class-name": "com.hazelcast.jet.examples.PersonSerializer"
 ```
 
-All the classes - data types, serializers & hooks - should be present
-on the cluster classpath, ideally in server's `lib` directory packaged as
-a jar file. Moreover, used type ids have to be unique across all
-serializers. Despite those limitations cluster level serializers offer
-full support for IMDG [sources and sinks](sources-sinks.md) - in particular
-the possibility to query & update `Map`s as well as read from `EventJournal`.
+Put the JAR containing the serializer hook and related classes in the
+`<jet_home>/lib` directory. Make sure that ecah registered serializer
+has a unique type ID.
+
+The advantage of a cluster-level serializer is that it is supported in
+all Hazelcast Jet features.
+
+## 3rd-Party Serialization Support
+
+### Google Protocol Buffers
+
+Since the classes generated by Google Protocol Buffers (Protobuf)
+already implement `java.io.Serializable`, Hazelcast Jet automatically
+supports them without a custom serializer. However, for best performance
+we encourage registering a Protobuf-specific serializer. There is a Jet
+extension module that simplifies this for Protobuf version 3.
+
+Add the extension as a dependency to your Jet job's Maven project:
+
+```xml
+<dependency>
+    <groupId>com.hazelcast.jet</groupId>
+    <artifactId>hazelcast-jet-protobuf</artifactId>
+    <classifier>jar-with-dependencies</classifier>
+    <version>${hazelcast.jet.version}</version>
+</dependency>
+```
+
+If you want to use it locally within a Jet job, implement an adapter
+by extending the provided class:
+
+```java
+class PersonSerializer extends ProtobufSerializer<Person> {
```

Comment:
wouldn't it be easier to use the registerSerializer(ProtobufSerializer.from(..) option?

Reply:
`registerSerializer()` takes serializer by class right now

Reply:
would it make sense to support instance as well?

Reply:
I guess we could do that, wanted to keep it simple.

Reply:
However, in case of Protobuf adapter which isn't `Serializable` (`Parser` is not `Serializable`) that would not be very useful.

Reply:
ok I see. 👍